src/main/clojure/aerospike_clj/client.clj

```diff
@@ -172,390 +105,367 @@
-           ^EventLoop (.next ^EventLoops (:el db))
-           (AsyncBatchListListener. op-future)
-           ^BatchPolicy (:policy conf)
-           ^List batch-reads-arr)
-     (let [d (p/chain op-future
-                      #(mapv batch-read->map %)
-                      (:transcoder conf identity))]
-      (register-events d db "read-batch" nil start-time)))))
-
-(defn exists-batch
-  "Check for each key in a batch if it exists. `indices` is a collection of maps
-  of the form `{:index \"key-name\" :set \"set-name\"}`. The result is a
-  vector of booleans in the same order as indices."
-  ([db indices]
-   (exists-batch db indices {}))
-  ([db indices conf]
-   (let [client (get-client db)
-         op-future (p/deferred)
-         start-time (System/nanoTime)
-         aero-namespace (:dbns db)
-         indices (utils/v->array Key (mapv #(create-key (:index %) aero-namespace (:set %)) indices))]
-     (.exists ^AerospikeClient client
-              ^EventLoop (.next ^EventLoops (:el db))
-              (AsyncExistsArrayListener. op-future)
-              ^BatchPolicy (:policy conf)
-              ^"[Lcom.aerospike.client.Key;" indices)
-     (let [d (p/chain op-future
-                       vec
-                       (:transcoder conf identity))]
-       (register-events d db "exists-batch" nil start-time)))))
-
-(defn get-multiple
-  "DEPRECATED - use `get-batch` instead.
-
-  Returns a (future) sequence of AerospikeRecords returned by `get-single`
-  with records in corresponding places to the required keys. Indices and sets should be sequences.
-  The `conf` map is passed to all `get-single` invocations."
-  {:deprecated "0.3.2"}
-  ([db indices sets]
-   (get-multiple db indices sets {}))
-  ([db indices sets conf]
-   (p/all
-     (map (fn [[index set-name]] (get-single db index set-name conf))
-          (map vector indices sets)))))
-
-(defn exists?
-  "Test if an index exists."
-  ([db index set-name] (exists? db index set-name {}))
-  ([db index set-name conf]
-   (let [client (get-client db index)
-         op-future (p/deferred)
-         start-time (System/nanoTime)]
-     (.exists ^AerospikeClient client
-              ^EventLoop (.next ^EventLoops (:el db))
-              (AsyncExistsListener. op-future)
-              ^Policy (:policy conf)
-              (create-key index (:dbns db) set-name))
-     (register-events op-future db "exists" index start-time))))
-
-(defn get-single-no-meta
-  "Shorthand to return a single record payload only."
-  ([db index set-name] (get-single db index set-name {:transcoder :payload}))
-  ([db index set-name ^IPersistentVector bin-names]
-   (get-single db index set-name {:transcoder :payload} bin-names)))
-
 ;; put
-(defn- _put [db index data policy set-name]
-  (let [client (get-client db index)
-        bins (data->bins data)
-        op-future (p/deferred)
+(defn- _put [^AerospikeClient client ^EventLoops event-loops dbns client-events index data policy set-name]
```

Comment:
A name that starts with an underscore means it is not being used and linters should ignore it.